cpp/src/io/text/multibyte_split.cu

```diff
@@ -160,6 +163,10 @@ struct PatternScan {
   }
 };
 
+// type aliases to distinguish between row offsets and character offsets
+using output_offset = int64_t;
+using byte_offset   = int64_t;
```

Comment:
Why is this? If you want to completely differentiate between these types, simple type aliasing like this is not enough. Instead, strong types should be prefered:
```
enum class output_offset : int64_t {};
enum class byte_offset : int64_t {};
```

Using strong types may be more difficult to do arithmetic operations but you can always cast the values to `int64_t` when doing so.

Reply:
Yes, this approach has its shortcomings, I mainly wanted to make clear what different offsets point to, more avoiding raw integer types for readability than true type safety. I need to to enough arithmetic on the types that using an enum class would be really inconvenient. Something like `BOOST_STRONG_TYPEDEF` would probably fit.

Reply:
Both approaches have pros and cons. For this case, I think a simple `using` is sufficient, with a weakly aliased type. It does pose some benefit to readability, and the need for arithmetic operators makes it difficult to use a strong type like an `enum class`. I'm happy with the tradeoffs that @upsj took in this case.

Reply:
Also: @ttnghia we _need_ strong types for the left/right comparator indices (the last time I remember discussing this topic) to ensure that function signatures match the given types. That isn't as much of a concern here because the weakly aliased types here aren't being passed to functions with multiple overloads for plain integers and strong types.